From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: kernel: Fix initialization of stp.
References: bnc#440610

Symptom:     The stp support cannot be activated.
Problem:     The return code of chsc_sstpc changed from 1 for success to
             0 for success. The stp_reset function still tests for 1 and
             does not set the indication that stp support is available.
Solution:    Check for the correct return code of chsc_sstpc.

Acked-by: John Jolly <jjolly@suse.de>
---
 arch/s390/kernel/time.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/s390/kernel/time.c
+++ b/arch/s390/kernel/time.c
@@ -1358,7 +1358,7 @@ static void __init stp_reset(void)
 
 	stp_page = alloc_bootmem_pages(PAGE_SIZE);
 	rc = chsc_sstpc(stp_page, STP_OP_CTRL, 0x0000);
-	if (rc == 1)
+	if (rc == 0)
 		set_bit(CLOCK_SYNC_HAS_STP, &clock_sync_flags);
 	else if (stp_online) {
 		pr_warning("The real or virtual hardware system does "
